hydra-main/src/main/java/com/addthis/hydra/job/alias/AliasManagerImpl.java

```diff
@@ -13,35 +13,83 @@
  */
 package com.addthis.hydra.job.alias;
 
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+import java.io.IOException;
+import java.io.StringWriter;
+
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
+import java.util.Objects;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.locks.ReentrantLock;
 
+import com.addthis.hydra.job.store.DataStoreUtil;
 import com.addthis.hydra.job.store.SpawnDataStore;
-import com.addthis.hydra.query.spawndatastore.AliasBiMap;
+import com.addthis.hydra.query.spawndatastore.AliasCache;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Strings;
+import com.google.common.base.Throwables;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.ObjectMapper;
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+@ThreadSafe
 public class AliasManagerImpl implements AliasManager {
-
     private static final Logger log = LoggerFactory.getLogger(AliasManagerImpl.class);
+    private Map<String, String> aliases;
+    /* This SpawnDataStore must be the same type (zookeeper/priam) between Spawn and Mqmaster. This should
+         * be guaranteed by the implementation of DataStoreUtil. */
+    public static final String ALIAS_PATH = "/query/alias";
+    private final SpawnDataStore spawnDataStore;
+    private final HashMap<String, List<String>> alias2jobs;
+    private final HashMap<String, String> job2alias;
```

Comment:
Why not use a [BiMap](https://google.github.io/guava/releases/19.0/api/docs/com/google/common/collect/BiMap.html)? That's cleaner IMO.